docs/source/pythonapi/index.rst

```diff
@@ -13,61 +13,261 @@ online. We recommend going through the modules from Codecademy_ and/or the
 `Scipy lectures`_. The full API documentation serves to provide more information
 on a given module or class.
 
-**Handling nuclear data:**
+------------------------------------
+:mod:`openmc` -- Basic Functionality
+------------------------------------
 
-.. toctree::
-    :maxdepth: 1
+Handling nuclear data
+---------------------
 
-    ace
-    mgxs_library
+Classes
++++++++
 
-**Creating input files:**
+.. autosummary::
+   :toctree: generated
+   :nosignatures:
+   :template: myclass.rst
 
-.. toctree::
-    :maxdepth: 1
+   openmc.XSdata
+   openmc.MGXSLibraryFile
 
-    cmfd
-    element
-    filter
-    geometry
-    material
-    mesh
-    nuclide
-    opencg_compatible
-    plots
-    settings
-    source
-    stats
-    surface
-    tallies
-    trigger
-    universe
-
-**Running OpenMC:**
+Functions
++++++++++
 
-.. toctree::
-    :maxdepth: 1
+.. autosummary::
+   :toctree: generated
+   :nosignatures:
 
-    executor
+   openmc.ace.ascii_to_binary
 
-**Post-processing:**
+Simulation Settings
+-------------------
 
-.. toctree::
-    :maxdepth: 1
+.. autosummary::
+   :toctree: generated
+   :nosignatures:
+   :template: myclass.rst
 
-    particle_restart
-    statepoint
-    summary
-    tallies
+   openmc.Source
+   openmc.ResonanceScattering
+   openmc.SettingsFile
 
-**Multi-Group Cross Section Generation**
+Material Specification
+----------------------
 
-.. toctree::
-    :maxdepth: 1
+.. autosummary::
+   :toctree: generated
+   :nosignatures:
+   :template: myclass.rst
+
+   openmc.Nuclide
+   openmc.Element
+   openmc.Macroscopic
+   openmc.Material
+   openmc.MaterialsFile
+
+Building geometry
+-----------------
+
+.. autosummary::
+   :toctree: generated
+   :nosignatures:
+   :template: myclass.rst
+
+   openmc.XPlane
+   openmc.YPlane
+   openmc.ZPlane
+   openmc.XCylinder
+   openmc.YCylinder
+   openmc.ZCylinder
+   openmc.Sphere
```

Comment:
Looks like we are missing `openmc.Cone` here? And it's subclasses `XCone`, `YCone` and `ZCone`.